sw/device/lib/dif/dif_usbdev.h

```diff
@@ -0,0 +1,759 @@
+ * - `kDifUsbdevInPacketSendResultGeneralError` if `endpoint` is invalid,
+ * - `kDifUsbdevInPacketSendResultNotWriting` if the `endpoint` does not have a
+ * packet to be sent,
+ * - `kDifUsbdevInPacketSendResultOK` otherwise.
+ */
+dif_usbdev_tx_packet_send_result_t dif_usbdev_tx_packet_send(
+    dif_usbdev_t *usbdev, uint32_t endpoint);
+
+/**
+ * Status of an outgoing packet.
+ */
+typedef enum dif_usbdev_tx_packet_status {
+  // There is no packet for the given endpoint
+  kDifUsbdevTxPacketStatusNoPacket,
+  // There is an active write buffer operation and the packet is not marked as
+  // ready for transmission yet
+  kDifUsbdevTxPacketStatusStillWriting,
+  // Packet is pending transmission
+  kDifUsbdevTxPacketStatusPending,
+  // Packet was sent successfully
+  kDifUsbdevTxPacketStatusSent,
+  // Transmission was canceled due to an incoming SETUP packet
+  kDifUsbdevTxPacketStatusCancelled,
+} dif_usbdev_tx_packet_status_t;
+
+/**
+ * Return codes for `dif_usbdev_tx_packet_get_status`.
+ */
+typedef enum dif_usbdev_tx_packet_get_status_result {
+  kDifUsbdevTxPacketGetStatusResultOK,
+  kDifUsbdevTxPacketGetStatusResultInvalidArgument,
+  kDifUsbdevTxPacketGetStatusResultGeneralError,
+} dif_usbdev_tx_packet_get_status_result_t;
+
+/**
+ * Get the status of a packet that has been queued to be sent from an endpoint.
+ *
+ * @param usbdev A USB device.
+ * @param endpoint An endpoint.
+ * @param status Status of the packet.
+ * @return
+ * - `kDifUsbdevTxPacketGetStatusResultInvalidArgument` if `usbdev` or
+ * `status` is `NULL`,
+ * - `kDifUsbdevTxPacketGetStatusResultGeneralError` if `endpoint` is invalid,
+ * - `kDifUsbdevTxPacketGetStatusResultOK` otherwise.
+ */
+dif_usbdev_tx_packet_get_status_result_t dif_usbdev_tx_packet_get_status(
+    dif_usbdev_t *usbdev, uint32_t endpoint,
+    dif_usbdev_tx_packet_status_t *status);
+
+/**
+ * Set the address of a USB device.
+ *
+ * @param usbdev A USB device.
+ * @param addr New address. Only the last 7 bits are significant.
+ * @return `kDifUsbdevResultInvalidArgument` if `usbdev` is `NULL`,
+ * `kDifUsbdevResultOK` otherwise.
+ */
+dif_usbdev_result_t dif_usbdev_address_set(dif_usbdev_t *usbdev, uint32_t addr);
+
+/**
+ * Get the address of a USB device.
+ *
+ * @param usbdev A USB device.
+ * @param addr Current address.
+ * @return `kDifUsbdevResultInvalidArgument` if `usbdev` or `addr` is `NULL`,
+ * `kDifUsbdevResultOK` otherwise.
+ */
+dif_usbdev_result_t dif_usbdev_address_get(dif_usbdev_t *usbdev,
+                                           uint32_t *addr);
+
+/**
+ * Clear the data toggle of an endpoint.
+ *
+ *
+ * @param usbdev A USB device.
+ * @param endpoint An endpoint.
+ * @return `kDifUsbdevResultInvalidArgument` if `usbdev` is `NULL`,
+ * `kDifUsbdevResultGeneralError` if `endpoint` is invalid,
+ * `kDifUsbdevResultOK` otherwise.
+ */
+dif_usbdev_result_t dif_usbdev_clear_data_toggle(dif_usbdev_t *usbdev,
```

Comment:
This functiontionality is for hw debug only (#2364) so it does not need to be included in the DIF. If needed it can go in a DV-specific extension.